### DISCUSSION TITLE
mpd + mpc on Raspberry Pi Bookworm : error when using file names with spaces

### DISCUSSION THREAD

**Discussion by Aldo-92** [MusicPlayerDaemon/MPD#2011] [Q&A]
I'm using mpd+mpc on a Raspberry Pi 4 and controlling it directly via terminal or via Python script with python-mpd2. This works fine, but I have a problem with file names containing spaces (ex.: my file.flac), even with the most basic sequence:
mpc add my file.flac
mpc play
==> "mpd.base.CommandError: [50@0] {add} no such directory"
I know that in a Linux terminal you should avoid spaces, because they are considered separators between parameters, but there are still ways to use them, for example with square brackets ("my file.flac") or backslash (my\ file.flac). These "tricks" (and others I have found on Internet) do not work with mpc command lines. I always get the same comment : "mpd.base.CommandError: [50@0] {add} no such directory".
Of course, one possibility would be to replace all spaces with underscores (my_file.flac works), but I would have to do that every time I buy music on a platform... Not really practical.
==> are there any other solutions for using filenames with spaces with mpd + mpc ?
Alain

**Comment by Aldo-92** [comment]
Problem solved. Two origins  : 
1. Raspberry Pi and Python were not using the same character set (utf-8 vs ISO8859-1)
2. I added client.update() to the script

**Reply by maxbobse** [reply]
Can you please share how you found out they are not using the same character set and what steps you did to fix it? Facing the same issue here.

**Reply by MaxKellermann** [reply]
Don't bother. This finding makes no sense.
